src/vs/workbench/services/configuration/common/jsonEditingService.ts

```diff
@@ -38,6 +39,18 @@ export class JSONEditingService implements IJSONEditingService {
 		return Promise.resolve(this.queue.queue(() => this.doWriteConfiguration(resource, values, save))); // queue up writes to prevent race conditions
 	}
 
+	// Throws if the content is not valid JSON
+	async read(resource: URI): Promise<any> {
+		const exists = await this.fileService.exists(resource);
+		if (!exists) {
+			throw new Error('Resource does not exist');
+		}
+		const reference = await this.textModelResolverService.createModelReference(resource);
+
+		const value = reference.object.textEditorModel.getValue();
+		return JSON.parse(stripComments(value));
```

Comment:
@bpasero if the argv.json file supports trailing `,` I can update this to use `parse` from `base/common/json`

Reply:
We read it very early in main.js where we dont have access to our modules:

https://github.com/microsoft/vscode/blob/32d40cf44e893e87ac33ac4f08de1e5f7fe077fc/src/main.js#L238-L246

Reply:
I think a big help would be to strip that trailing comma on the last line of `argv.json` if one is there. Logic for that shouldn't be too bad right? What do you think?

Reply:
Martin did this already.